packages/gatsby-react-router-scroll/src/scroll-container.tsx

```diff
@@ -23,9 +25,25 @@ interface IPropsWithContextAndLocation extends IProps {
   location: HLocation
 }
 
+let hasNotWarnedDeprecation = true
+
 class ScrollContainerImplementation extends React.Component<
   IPropsWithContextAndLocation
 > {
+  constructor(props) {
+    super(props)
+
+    if (hasNotWarnedDeprecation) {
```

Comment:
Only show deprecations on dev mode
```suggestion
    if (process.env.NODE_ENV !== 'production' && hasNotWarnedDeprecation) {
```